core/toolbox/toolbox.js

```diff
@@ -13,56 +13,54 @@
 goog.module('Blockly.Toolbox');
 goog.module.declareLegacyNamespace();
 
-goog.require('Blockly');
-goog.require('Blockly.BlockSvg');
-goog.require('Blockly.browserEvents');
-goog.require('Blockly.CollapsibleToolboxCategory');
-goog.require('Blockly.ComponentManager');
-goog.require('Blockly.Css');
-goog.require('Blockly.DeleteArea');
-goog.require('Blockly.Events');
+const Blockly = goog.require('Blockly');
+const BlocklyOptions = goog.requireType('Blockly.BlocklyOptions');
+const BlockSvg = goog.require('Blockly.BlockSvg');
+const CollapsibleToolboxCategory = goog.require('Blockly.CollapsibleToolboxCategory');
+const ComponentManager = goog.require('Blockly.ComponentManager');
+const Css = goog.require('Blockly.Css');const DeleteArea = goog.require('Blockly.DeleteArea');
+const Events = goog.require('Blockly.Events');
+const IAutoHideable = goog.require('Blockly.IAutoHideable');
```

Comment:
Should this not be `requireType`?  (Here and several more interfaces below.)

Reply:
Interfaces that are implemented need to be `require`'d explicitely.

Reply:
I'm not aware of any such requirement.  Can you direct me to more information about this?  As far as I knew interfaces are *nothing but* a type—the only code is an empty `function () {}`—so the interface name should only appear in comments and so there ought not to be any reason to `require` it.

(Things are different in TypeScript, of course, where interface names appear outside of comments.)

Reply:
After a bit more experimentation I'm sufficiently confident that you are mistaken about this to open PR #5343—but I have taken the precaution of assigning it to you for review, since it is quite possible that there are concerns beyond "does it compile and pass tests" that I am not aware of.